### PR TITLE
Make profile settings save button sticky when modified

### PR DESCRIPTION
**Pull request by emgoto**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

When a user modifies a field on the profile settings page, the save button footer will become "sticky" along the bottom of the screen (see GIF below).

## Related Tickets & Documents

This is for issue #9240 

## QA Instructions, Screenshots, Recordings
sticky

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

~Happy to add a test - just not sure how or where!~ (contributor added tests)

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/javascript/packs/stickySaveFooter.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const form = document.querySelector('.edit_user');
```
Comment (rhymes) [inline app/javascript/packs/stickySaveFooter.js:1]: missing the newline at the end of the file

File: app/javascript/packs/stickySaveFooter.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+form.addEventListener('change', () => {
+    const saveFooter = document.querySelector('#save-footer');
+    saveFooter.classList.add('sticky-save-footer');
```
Comment (rhymes) [inline app/javascript/packs/stickySaveFooter.js:5]: to be more robust, always check if the element is there after querying it and know that `getElementByClassName` is much faster than `querySelector`: https://jsperf.com/getelementsbyclassname-0-vs-queryselector/213

File: app/assets/stylesheets/settings.scss [hunk 3]
```diff
@@ -139,3 +139,9 @@ $input-width: 650px;
 .crayons-notice--danger h3 {
   color: var(--accent-danger-darker);
 }
+
+.sticky-save-footer {
```
Comment (rhymes) [inline app/assets/stylesheets/settings.scss:143]: missing the new line at the end of the file

**Comment by emgoto** [comment]
Thanks @rhymes! I made the changes you suggested. I'm surprised the precommit hook didn't pick up on the new line stuff.

**Comment by rhymes** [comment]
@emgoto 

> I have a PR etiquette question - is it considered polite/normal to use the "re-request review" feature? And am I okay to close off your comments as I address them? (I usually use Bitbucket for code reviews so some of this is new to me).

definitely yes to both :) 

> Edit: Another question, when a contributor has an issue assigned to them but there's no activity on it, is there a certain amount of time we can let pass before we double-check if we can pick up the issue instead? (A month?)

We don't have a set amount of time but if you find an issue you'd like to take over and see the maintainer hasn't responded in a while please leave a comment so we can reassign it.

Thank you!

**Comment by rhymes** [comment]
@emgoto I rebased your commits on top of master and force pushed to fix the failing flaky tests

**Comment by ludwiczakpawel** [comment]
Hey @emgoto I know it's merged already but just wanted to say "thanks"! Really really nice 👍 

Btw. do you think we could add this for other settings pages as well? At least for /settings/ux since this one, I think, is visited and updated pretty often :)

**Comment by emgoto** [comment]
Yes, that would be cool! Do we need a separate issue for it, or I can just raise a PR against the same issue (#9240)?

**Comment by ludwiczakpawel** [comment]
@emgoto no need for separate issue, just a PR would be enough :)

**Comment by emgoto** [comment]
I've raised the PR for `settings/ux`. I've chosen a theme in the past and navigated away, not realising there was even a save button so hopefully this will help other people from making that same mistake!

I think for any other pages that we want to add this to, it would make a great candidate for being an issue tagged with `good first issue`, as any new devs would be able to check out my PR to see how to do it.

**Pull request merged**